### DISCUSSION TITLE
Not able to connect after upgrading to 5.9.14

### DISCUSSION THREAD

**Discussion by vomus** [strongswan/strongswan#2312] [Q&A]
Hi!
I just upgraded strongswan from 5.9.13 + NM 1.5.2 to 5.9.14 + NM 1.6.0 and found out that it stopped connecting with a preshared key. Downgrading to older versions healed the problem. The log is below. It does not seem to a NetworkManager issue, however... 

Jun 19 09:05:38 yarilo NetworkManager[1278]: <info>  [1718777138.1664] audit: op="statistics" interface="eth2" ifindex=3 args="2000" pid=21511 uid=500 result="success"
Jun 19 09:05:38 yarilo NetworkManager[1278]: <info>  [1718777138.1708] audit: op="statistics" interface="wlan0" ifindex=2 args="2000" pid=21511 uid=500 result="success"
Jun 19 09:05:40 yarilo plasmashell[21511]: QDBusObjectPath: invalid path ""
Jun 19 09:05:40 yarilo NetworkManager[1278]: <info>  [1718777140.6779] vpn[0x55e06bc88d70,f2bc4893-9323-419a-9405-7e1fb0d4b86e,"Safian"]: starting l2tp
Jun 19 09:05:40 yarilo NetworkManager[1278]: <info>  [1718777140.6786] audit: op="connection-activate" uuid="f2bc4893-9323-419a-9405-7e1fb0d4b86e" name="Safian" pid=21511 uid=500 result="success"
Jun 19 09:05:40 yarilo kded5[21463]: org.kde.plasma.nm.kded: Unhandled VPN connection state change:  NetworkManager::VpnConnection::NeedAuth
Jun 19 09:05:40 yarilo kded5[21463]: kf.networkmanagerqt: void NetworkManager::ConnectionPrivate::onPropertiesChanged(const QVariantMap&) Unhandled property "VersionId"
Jun 19 09:05:40 yarilo kded5[21463]: org.kde.plasma.nm.kded: Unhandled VPN connection state change:  NetworkManager::VpnConnection::Connecting
Jun 19 09:05:40 yarilo plasmashell[21511]: kf.networkmanagerqt: void NetworkManager::ConnectionPrivate::onPropertiesChanged(const QVariantMap&) Unhandled property "VersionId"
Jun 19 09:05:40 yarilo nm-l2tp-service[22667]: Check port 1701
Jun 19 09:05:40 yarilo NetworkManager[22681]: Stopping strongSwan IPsec failed: starter is not running
Jun 19 09:05:42 yarilo NetworkManager[22678]: Starting strongSwan 5.9.14 IPsec [starter]...
Jun 19 09:05:42 yarilo ipsec_starter[22678]: Starting strongSwan 5.9.14 IPsec [starter]...
Jun 19 09:05:42 yarilo NetworkManager[22678]: Loading config setup
Jun 19 09:05:42 yarilo NetworkManager[22678]: Loading conn 'f2bc4893-9323-419a-9405-7e1fb0d4b86e'
Jun 19 09:05:42 yarilo ipsec_starter[22678]: Loading config setup
Jun 19 09:05:42 yarilo ipsec_starter[22678]: Loading conn 'f2bc4893-9323-419a-9405-7e1fb0d4b86e'
Jun 19 09:05:42 yarilo ipsec_starter[22699]: Attempting to start charon...
Jun 19 09:05:42 yarilo charon[22700]: 00[DMN] Starting IKE charon daemon (strongSwan 5.9.14, Linux 6.6.33-un-def-alt1, x86_64)
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] PKCS11 module '<name>' lacks library path
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] providers loaded by OpenSSL: legacy default
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'sqlite': failed to load - sqlite_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] created TUN device: ipsec0
Jun 19 09:05:42 yarilo kernel: tun: Universal TUN/TAP device driver, 1.6
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] using '/sbin/resolvconf' to install DNS servers
Jun 19 09:05:42 yarilo NetworkManager[1278]: <info>  [1718777142.8510] manager: (ipsec0): new Tun device (/org/freedesktop/NetworkManager/Devices/5)
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'eap-tnc': failed to load - eap_tnc_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo avahi-daemon[1134]: Joining mDNS multicast group on interface ipsec0.IPv6 with address fe80::331b:9967:7aad:748c.
Jun 19 09:05:42 yarilo avahi-daemon[1134]: New relevant interface ipsec0.IPv6 for mDNS.
Jun 19 09:05:42 yarilo avahi-daemon[1134]: Registering new address record for fe80::331b:9967:7aad:748c on ipsec0.*.
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'tnc-ifmap': failed to load - tnc_ifmap_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'tnc-pdp': failed to load - tnc_pdp_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'tnc-imc': failed to load - tnc_imc_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'tnc-imv': failed to load - tnc_imv_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'tnc-tnccs': failed to load - tnc_tnccs_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'tnccs-20': failed to load - tnccs_20_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'tnccs-11': failed to load - tnccs_11_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] plugin 'tnccs-dynamic': failed to load - tnccs_dynamic_plugin_create not found and no plugin file available
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] loading ca certificates from '/etc/strongswan/ipsec.d/cacerts'
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] loading aa certificates from '/etc/strongswan/ipsec.d/aacerts'
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] loading ocsp signer certificates from '/etc/strongswan/ipsec.d/ocspcerts'
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] loading attribute certificates from '/etc/strongswan/ipsec.d/acerts'
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] loading crls from '/etc/strongswan/ipsec.d/crls'
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] loading secrets from '/etc/strongswan/ipsec.secrets'
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] loading secrets from '/etc/strongswan/ipsec.d/ipsec.nm-l2tp.secrets'
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG]   loaded IKE secret for %any
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] opening triplet file /etc/strongswan/ipsec.d/triplets.dat failed: No such file or directory
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] loaded 0 RADIUS server configurations
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] HA config misses local/remote address
Jun 19 09:05:42 yarilo charon[22700]: 00[CFG] no script for ext-auth script defined, disabled
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] loaded plugins: charon ldap pkcs11 tpm aesni aes des rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints acert pubkey pkcs1 pkcs7 pkcs12 pgp dnskey sshkey pem openssl gcrypt pkcs8 fips-prf gmp curve25519 agent chapoly xcbc cmac hmac kdf ctr ccm gcm ntru drbg newhope curl attr kernel-libipsec kernel-netlink resolve socket-default farp stroke vici smp updown eap-identity eap-sim eap-sim-pcsc eap-aka eap-aka-3gpp eap-aka-3gpp2 eap-md5 eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls eap-peap xauth-generic xauth-eap xauth-pam xauth-noauth dhcp led duplicheck addrblock unity counters
Jun 19 09:05:42 yarilo charon[22700]: 00[LIB] dropped capabilities, running as uid 0, gid 0
Jun 19 09:05:42 yarilo charon[22700]: 00[JOB] spawning 16 worker threads
Jun 19 09:05:42 yarilo ipsec_starter[22699]: charon (22700) started after 40 ms
Jun 19 09:05:42 yarilo charon[22700]: 11[CFG] received stroke: add connection 'f2bc4893-9323-419a-9405-7e1fb0d4b86e'
Jun 19 09:05:42 yarilo charon[22700]: 11[CFG] added configuration 'f2bc4893-9323-419a-9405-7e1fb0d4b86e'
Jun 19 09:05:43 yarilo charon[22700]: 11[CFG] rereading secrets
Jun 19 09:05:43 yarilo charon[22700]: 11[CFG] loading secrets from '/etc/strongswan/ipsec.secrets'
Jun 19 09:05:43 yarilo charon[22700]: 11[CFG] loading secrets from '/etc/strongswan/ipsec.d/ipsec.nm-l2tp.secrets'
Jun 19 09:05:43 yarilo charon[22700]: 11[CFG]   loaded IKE secret for %any
Jun 19 09:05:43 yarilo charon[22700]: 13[CFG] received stroke: initiate 'f2bc4893-9323-419a-9405-7e1fb0d4b86e'
Jun 19 09:05:43 yarilo charon[22700]: 15[IKE] initiating Main Mode IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1] to xx.xxx.xx.xxx
Jun 19 09:05:43 yarilo charon[22700]: 15[IKE] initiating Main Mode IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1] to xx.xxx.xx.xxx
Jun 19 09:05:43 yarilo charon[22700]: 15[ENC] generating ID_PROT request 0 [ SA V V V V V ]
Jun 19 09:05:43 yarilo charon[22700]: 15[NET] sending packet: from yyy.yy.yyy.yy[500] to xx.xxx.xx.xxx[500] (216 bytes)
Jun 19 09:05:43 yarilo charon[22700]: 01[NET] received packet: from xx.xxx.xx.xxx[500] to yyy.yy.yyy.yy[500] (160 bytes)
Jun 19 09:05:43 yarilo charon[22700]: 01[ENC] parsed ID_PROT response 0 [ SA V V V V ]
Jun 19 09:05:43 yarilo charon[22700]: 01[IKE] received NAT-T (RFC 3947) vendor ID
Jun 19 09:05:43 yarilo charon[22700]: 01[IKE] received XAuth vendor ID
Jun 19 09:05:43 yarilo charon[22700]: 01[IKE] received DPD vendor ID
Jun 19 09:05:43 yarilo charon[22700]: 01[IKE] received FRAGMENTATION vendor ID
Jun 19 09:05:43 yarilo charon[22700]: 01[CFG] selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
Jun 19 09:05:43 yarilo charon[22700]: 01[ENC] generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
Jun 19 09:05:43 yarilo charon[22700]: 01[NET] sending packet: from yyy.yy.yyy.yy[500] to xx.xxx.xx.xxx[500] (244 bytes)
Jun 19 09:05:44 yarilo charon[22700]: 12[NET] received packet: from xx.xxx.xx.xxx[500] to yyy.yy.yyy.yy[500] (236 bytes)
Jun 19 09:05:44 yarilo charon[22700]: 12[ENC] parsed ID_PROT response 0 [ KE No NAT-D NAT-D ]
Jun 19 09:05:44 yarilo charon[22700]: 12[IKE] local host is behind NAT, sending keep alives
Jun 19 09:05:44 yarilo charon[22700]: 12[ENC] generating ID_PROT request 0 [ ID HASH ]
Jun 19 09:05:44 yarilo charon[22700]: 12[NET] sending packet: from yyy.yy.yyy.yy[4500] to xx.xxx.xx.xxx[4500] (76 bytes)
Jun 19 09:05:44 yarilo charon[22700]: 14[NET] received packet: from xx.xxx.xx.xxx[4500] to yyy.yy.yyy.yy[4500] (76 bytes)
Jun 19 09:05:44 yarilo charon[22700]: 14[ENC] parsed ID_PROT response 0 [ ID HASH ]
Jun 19 09:05:44 yarilo charon[22700]: 14[IKE] IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1] established between yyy.yy.yyy.yy[yyy.yy.yyy.yy]...xx.xxx.xx.xxx[xx.xxx.xx.xxx]
Jun 19 09:05:44 yarilo charon[22700]: 14[IKE] IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1] established between yyy.yy.yyy.yy[yyy.yy.yyy.yy]...xx.xxx.xx.xxx[xx.xxx.xx.xxx]
Jun 19 09:05:44 yarilo charon[22700]: 14[IKE] scheduling reauthentication in 10238s
Jun 19 09:05:44 yarilo charon[22700]: 14[IKE] maximum IKE_SA lifetime 10778s
Jun 19 09:05:44 yarilo charon[22700]: 14[ENC] generating QUICK_MODE request 827676382 [ HASH SA No ID ID NAT-OA NAT-OA ]
Jun 19 09:05:44 yarilo charon[22700]: 14[NET] sending packet: from yyy.yy.yyy.yy[4500] to xx.xxx.xx.xxx[4500] (204 bytes)
Jun 19 09:05:44 yarilo charon[22700]: 15[NET] received packet: from xx.xxx.xx.xxx[4500] to yyy.yy.yyy.yy[4500] (188 bytes)
Jun 19 09:05:44 yarilo charon[22700]: 15[ENC] parsed QUICK_MODE response 827676382 [ HASH SA No ID ID NAT-OA NAT-OA ]
Jun 19 09:05:44 yarilo charon[22700]: 15[CFG] selected proposal: ESP:AES_CBC_128/HMAC_SHA1_96/NO_EXT_SEQ
Jun 19 09:05:44 yarilo charon[22700]: 15[ESP]   IPsec SA: unsupported mode
Jun 19 09:05:44 yarilo charon[22700]: 15[ESP] failed to create SAD entry
Jun 19 09:05:44 yarilo NetworkManager[22738]: initiating Main Mode IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1] to xx.xxx.xx.xxx
Jun 19 09:05:44 yarilo NetworkManager[22738]: generating ID_PROT request 0 [ SA V V V V V ]
Jun 19 09:05:44 yarilo NetworkManager[22738]: sending packet: from yyy.yy.yyy.yy[500] to xx.xxx.xx.xxx[500] (216 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22738]: received packet: from xx.xxx.xx.xxx[500] to yyy.yy.yyy.yy[500] (160 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22738]: parsed ID_PROT response 0 [ SA V V V V ]
Jun 19 09:05:44 yarilo NetworkManager[22738]: received NAT-T (RFC 3947) vendor ID
Jun 19 09:05:44 yarilo NetworkManager[22738]: received XAuth vendor ID
Jun 19 09:05:44 yarilo NetworkManager[22738]: received DPD vendor ID
Jun 19 09:05:44 yarilo NetworkManager[22738]: received FRAGMENTATION vendor ID
Jun 19 09:05:44 yarilo NetworkManager[22738]: selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
Jun 19 09:05:44 yarilo NetworkManager[22738]: generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
Jun 19 09:05:44 yarilo NetworkManager[22738]: sending packet: from yyy.yy.yyy.yy[500] to xx.xxx.xx.xxx[500] (244 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22738]: received packet: from xx.xxx.xx.xxx[500] to yyy.yy.yyy.yy[500] (236 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22738]: parsed ID_PROT response 0 [ KE No NAT-D NAT-D ]
Jun 19 09:05:44 yarilo NetworkManager[22738]: local host is behind NAT, sending keep alives
Jun 19 09:05:44 yarilo NetworkManager[22738]: generating ID_PROT request 0 [ ID HASH ]
Jun 19 09:05:44 yarilo NetworkManager[22738]: sending packet: from yyy.yy.yyy.yy[4500] to xx.xxx.xx.xxx[4500] (76 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22738]: received packet: from xx.xxx.xx.xxx[4500] to yyy.yy.yyy.yy[4500] (76 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22738]: parsed ID_PROT response 0 [ ID HASH ]
Jun 19 09:05:44 yarilo NetworkManager[22738]: IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1] established between yyy.yy.yyy.yy[yyy.yy.yyy.yy]...xx.xxx.xx.xxx[xx.xxx.xx.xxx]
Jun 19 09:05:44 yarilo NetworkManager[22738]: scheduling reauthentication in 10238s
Jun 19 09:05:44 yarilo NetworkManager[22738]: maximum IKE_SA lifetime 10778s
Jun 19 09:05:44 yarilo NetworkManager[22738]: generating QUICK_MODE request 827676382 [ HASH SA No ID ID NAT-OA NAT-OA ]
Jun 19 09:05:44 yarilo NetworkManager[22738]: sending packet: from yyy.yy.yyy.yy[4500] to xx.xxx.xx.xxx[4500] (204 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22738]: received packet: from xx.xxx.xx.xxx[4500] to yyy.yy.yyy.yy[4500] (188 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22738]: parsed QUICK_MODE response 827676382 [ HASH SA No ID ID NAT-OA NAT-OA ]
Jun 19 09:05:44 yarilo NetworkManager[22738]: selected proposal: ESP:AES_CBC_128/HMAC_SHA1_96/NO_EXT_SEQ
Jun 19 09:05:44 yarilo NetworkManager[22738]:   IPsec SA: unsupported mode
Jun 19 09:05:44 yarilo NetworkManager[22738]: failed to create SAD entry
Jun 19 09:05:44 yarilo NetworkManager[22738]:   IPsec SA: unsupported mode
Jun 19 09:05:44 yarilo NetworkManager[22738]: failed to create SAD entry
Jun 19 09:05:44 yarilo NetworkManager[22738]: unable to install inbound and outbound IPsec SA (SAD) in kernel
Jun 19 09:05:44 yarilo NetworkManager[22738]: establishing connection 'f2bc4893-9323-419a-9405-7e1fb0d4b86e' failed
Jun 19 09:05:44 yarilo charon[22700]: 15[ESP]   IPsec SA: unsupported mode
Jun 19 09:05:44 yarilo charon[22700]: 15[ESP] failed to create SAD entry
Jun 19 09:05:44 yarilo charon[22700]: 15[IKE] unable to install inbound and outbound IPsec SA (SAD) in kernel
Jun 19 09:05:44 yarilo charon[22700]: 15[ENC] generating INFORMATIONAL_V1 request 1296478391 [ HASH N(NO_PROP) ]
Jun 19 09:05:44 yarilo charon[22700]: 15[NET] sending packet: from yyy.yy.yyy.yy[4500] to xx.xxx.xx.xxx[4500] (76 bytes)
Jun 19 09:05:44 yarilo NetworkManager[22745]: Stopping strongSwan IPsec...
Jun 19 09:05:44 yarilo charon[22700]: 00[DMN] SIGINT received, shutting down
Jun 19 09:05:44 yarilo charon[22700]: 00[IKE] deleting IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1] between yyy.yy.yyy.yy[yyy.yy.yyy.yy]...xx.xxx.xx.xxx[xx.xxx.xx.xxx]
Jun 19 09:05:44 yarilo charon[22700]: 00[IKE] deleting IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1] between yyy.yy.yyy.yy[yyy.yy.yyy.yy]...xx.xxx.xx.xxx[xx.xxx.xx.xxx]
Jun 19 09:05:44 yarilo charon[22700]: 00[IKE] sending DELETE for IKE_SA f2bc4893-9323-419a-9405-7e1fb0d4b86e[1]
Jun 19 09:05:44 yarilo charon[22700]: 00[ENC] generating INFORMATIONAL_V1 request 647635 [ HASH D ]
Jun 19 09:05:44 yarilo charon[22700]: 00[NET] sending packet: from yyy.yy.yyy.yy[4500] to xx.xxx.xx.xxx[4500] (92 bytes)
Jun 19 09:05:44 yarilo avahi-daemon[1134]: Interface ipsec0.IPv6 no longer relevant for mDNS.
Jun 19 09:05:44 yarilo avahi-daemon[1134]: Leaving mDNS multicast group on interface ipsec0.IPv6 with address fe80::331b:9967:7aad:748c.
Jun 19 09:05:44 yarilo avahi-daemon[1134]: Withdrawing address record for fe80::331b:9967:7aad:748c on ipsec0.
Jun 19 09:05:44 yarilo ipsec_starter[22699]: child 22700 (charon) has quit (exit code 0)
Jun 19 09:05:44 yarilo ipsec_starter[22699]:
Jun 19 09:05:44 yarilo ipsec_starter[22699]: charon stopped after 200 ms
Jun 19 09:05:44 yarilo ipsec_starter[22699]: ipsec starter stopped
Jun 19 09:05:44 yarilo nm-l2tp-service[22667]: Could not establish IPsec connection.

**Comment by tobiasbrunner** [comment]
You have the [`kernel-libipsec` plugin](https://docs.strongswan.org/docs/5.9/plugins/kernel-libipsec.html) loaded for some reason. You  don't want that. In your case specifically because the plugin does not support transport mode:

> Jun 19 09:05:44 yarilo charon[22700]: 15[ESP] IPsec SA: unsupported mode
